src/Microsoft.DotNet.ProjectModel/Compilation/LibraryExporter.cs

```diff
@@ -91,16 +93,14 @@ private IEnumerable<LibraryExport> ExportLibraries(Func<LibraryDescription, bool
                     }
                 }
 
+                // Source references are not transitive
```

Comment:
God no depth level, make it non transitive by default and we can implement the nuget feature for flowing things.